mlir/lib/Gradient/Transforms/BufferizationPatterns.cpp

```diff
@@ -32,8 +34,25 @@ class BufferizeAdjointOp : public OpConversionPattern<AdjointOp> {
         SmallVector<Type> resTypes;
         if (failed(getTypeConverter()->convertTypes(op.getResultTypes(), resTypes)))
             return failure();
-        rewriter.replaceOpWithNewOp<AdjointOp>(op, resTypes, op.getCalleeAttr(),
-                                               adaptor.getOperands());
+
+        Location loc = op.getLoc();
+        Value gradSize = op.getGradSize();
+        SmallVector<Value> memrefValues;
+        SmallVector<Value> tensorValues;
+        int i = 0;
+        for (Type resType : resTypes) {
+            MemRefType memrefType = resType.cast<MemRefType>();
+            Value memrefValue = rewriter.create<memref::AllocOp>(loc, memrefType, gradSize);
+            Value tensorValue = rewriter.create<bufferization::ToTensorOp>(
+                loc, op.getResultTypes()[i++], memrefValue);
+            memrefValues.push_back(memrefValue);
+            tensorValues.push_back(tensorValue);
+        }
```

Comment:
I think the converter automatically inserts unrealized casts for us :) 
```suggestion
        for (Type resType : resTypes) {
            MemRefType memrefType = resType.cast<MemRefType>();
            Value memrefValue = rewriter.create<memref::AllocOp>(loc, memrefType, gradSize);
            memrefValues.push_back(memrefValue);
        }
```

Reply:
[66b8310](https://github.com/PennyLaneAI/catalyst/pull/63/commits/66b831097c5b40dd6f81a894dbe50ac04d8ccf7e)